CHANGELOG.md

```diff
@@ -14,6 +14,11 @@ as necessary. Empty sections will not end in the release notes.
 
 ### New Features
 
+* Introduces a hard objects-cache capacity limit to ensure that the cache does never consume more than
+  the configured cache-capacity plus a configurable "overshoot" (defaults to 10%). New cache entries are
+  admitted as long as the current cache size is less than the "cache-capacity + overshoot". The
+  "overshoot" is necessary to eventually trigger the cache eviction.
```

Comment:
I'm not sure the last sentence is 100% correct. From my POV the overshoot is needed for Nessie code to admit new entries when the cache is full (because eviction will not reduce the cache much lower than the limit). Eviction will happen, of course, but it's not the reason for having the overshoot parameter.